lib/ip_blocklist.js

```
@@ -0,0 +1,96 @@
+  }
+
+  function parseRow(ipColumn) {
+    // Ignore # lines
+    if (ipColumn.length > 0 && ipColumn[0] === '#') {
+      return
+    }
+
+    // Parse row for ip address components, supports 1.10.16.0/20 or 1.10.16.0.
+    // Converts them to a base IP address and a prefix length.
+    var tokens = ipColumn.split('/')
+    if (tokens.length === 2) {
+      var prefixLength = parseInt(tokens[1], 10)
+      return {
+        baseIp: getBaseIp(tokens[0], prefixLength),
+        prefixLength: prefixLength
+      }
+    }
+
+    return {
+      baseIp: tokens[0],
+      prefixLength: 32
+    }
+  }
+
+  function IPBlocklist() {
+    this.prefixLengths = []
+    this.ipsByPrefixLength = {}
+  }
+
+  IPBlocklist.prototype.load = function (csvPath) {
+    var self = this
+    csvPath = path.join(__dirname, csvPath)
+
+    return readFile(csvPath, 'utf8')
+      .then(function (data) {
+        return parse(data)
+      })
+      .then(function (rows) {
+        rows.forEach(function (row) {
+          var parsedData = parseRow(row[0])
+
+          if (parsedData) {
+            if (! self.ipsByPrefixLength[parsedData.prefixLength]) {
+              self.prefixLengths.push(parsedData.prefixLength)
+              self.ipsByPrefixLength[parsedData.prefixLength] = {}
+            }
+            self.ipsByPrefixLength[parsedData.prefixLength][parsedData.baseIp] = true
+          }
+        })
+      })
+  }
+
+  IPBlocklist.prototype.clear = function () {
+    this.prefixLengths = []
+    this.ipsByPrefixLength = {}
+  }
+
+  IPBlocklist.prototype.contains = function (ipAddress) {
+    var self = this
+
+    try {
+      return self.prefixLengths.some(function (prefixLength) {
+        return getBaseIp(ipAddress, prefixLength) in self.ipsByPrefixLength[prefixLength]
+      })
+    } catch(err){
+      // On any error, side on caution and allow the request through. These errors could be
+      // malformed ip address, etc
```

In other places in customs-server, we deliberately fail closed for security.  It's probably worth being consistent here.